labs/fabric-smart-client.md

```diff
@@ -0,0 +1,60 @@
+
+Hyperledger Fabric Smart Client
+
+# Short Description
+
+[Hyperledger Fabric]('https://www.hyperledger.org/use/fabric') is a permissioned, modular, and extensible open-source DLT platform. Fabric architecture follows a novel `execute-order-validate` paradigm that supports distributed execution of untrusted code in an untrusted environment. Indeed, Fabric-based distributed applications can be written in any general-purpose programming language.  
+
+Developing applications for Hyperledger Fabric is often hard, sometimes painful. Fabric is a very powerful ecosystem whose building blocks must be carefully orchestrated to achieve the desired results. Currently, the Fabric Client SDKs are too limited. They do not offer any advanced capabilities to let the developers focus on the `application business processes`, and harness the full potential of Fabric.
+
+What would happen if the developers could use a `Smart(er) Fabric Client` that offers:
+- A high level API that hides the complexity of Fabric;
+- A Peer-to-Peer infrastructure that let business parties talk to each as required by the business processes;
+- An advanced transaction orchestration mechanism;
+- A simplified model to interact with chaincodes;
+- A local storage place where to store the states the application cares about;
+- A State-based programming model that let you forget about RW sets and focus on business objects and their interactions?
+Developing Fabric-based distributed applications would become simpler and joyful.
+If you are a domain expert, the Fabric Smart Client hides the complexity of Fabric and allows you to focus on the business interactions.
+If you are a skilled software engineer, you will be able to leverage the full power of Fabric.
+
+# Scope of Lab
+
+The scope of this lab is to deliver the `Fabric Smart Client` (FSC, for short), the `Next-Generation Fabric Client SDK`. FSC simplifies the development of Fabric-based distributed applications by putting at the centre the `business processes`, and by `hiding the unnecessary burden` of dealing with Fabric.
+
+A `business process` is a collection of `related, structured activities` carried by `business parties` in which a specific sequence of steps serves a particular `business goal`. In other words, a business process is an `interactive protocol` among `business parties` each of which has its own `business view` (or simply `view`) of the process. A `view` is then the sequence of steps that a business party must execute to serve a particular `business goal`.
+FSC aims to offer a comfortable environment to build such interactive protocols backed by the Hyperledger Fabric blockchain.
+
+FSC will consist of multiple SDKs each of which provides a set of API and services that can be either used directly by FSC developers, to build their applications, or by other SDKs. A network node that runs the FSC stack is called `FSC node`. FSC nodes form an `FSC network` that will leave on a side of one or more Fabric networks.
```

Comment:
is `will leave` supposed to be `will live`?

Reply:
yeah, it's got to be. :)

Reply:
fixed, thanks :)